kernel/src/main/java/org/kframework/kompile/DefinitionParsing.java

```diff
@@ -111,23 +113,36 @@ public DefinitionParsing(
                 kore,
                 options.preprocess);
 
+        if (!def.getModule(mainModule).isDefined()) {
+          throw KEMException.criticalError("Module " + mainModule + " does not exist.");
+        }
+        if (!def.getModule(entryPointModule).isDefined()) {
+          throw KEMException.criticalError("Module " + entryPointModule + " does not exist.");
+        }
+        Stream<Module> modules = Stream.of(def.getModule(mainModule).get());
+        modules = Stream.concat(modules, stream(def.getModule(mainModule).get().importedModules()));
+        modules = Stream.concat(modules, Stream.of(def.getModule(entryPointModule).get()));
+        modules = Stream.concat(modules, stream(def.getModule(entryPointModule).get().importedModules()));
+        modules = Stream.concat(modules,
+                stream(def.entryModules()).filter(m -> !stream(m.sentences()).anyMatch(s -> s instanceof Bubble)));
+        def = Definition(def.mainModule(), modules.collect(Collections.toSet()), def.att());
```

Comment:
I don't think I fully understand this, so want to make sure I do.

We're grabbing the entryPoint and main modules, as well as any modules they transitively import. We're also grabbing any entryModule (which I guess would be something like the proving module?), but _not_ the transitive imports of the entryModules? And we're specifically only grabbing entryModules which have bubbles which need parsing.

I guess my questions are:

-   Why not get transitive imports of entryModules()?
-   Why not apply the filter for only modules with bubbles across _all_ the modules, not just the entryModules()?

Reply:
so, Definition.entryModules is a bit of a misnomer insofar as it contains every module that was included in the K files you parsed. that's why we're not doing a transitive closure there. And specifically we are only grabbing those modules which /don't/ contain bubbles. And the only reason we are doing this is because some of them are required to exist in the parser, like some of the ones in kast.k.

Reply:
Ok, so it's an over-approximation of the modules needed to do parsing.

Reply:
1. I recommend a comment explaining the above logic. Lack of comments in such places is a reason why code is hard to maintain. 
2. I don't see any advantage to build a Stream then to convert it to Set. You should create a Set from start. Code will look simpler. If you want the preserve the iterating order then use `LinkedHashSet`.